File: 0167-two-sum-ii---input-array-is-sorted/solution.go
func twoSum(numbers []int, target int) []int {
    left := 0
    right := len(numbers) - 1

    for left < len(numbers) - 1 {
        sum := numbers[left] + numbers[right]
        if sum == target {
            return []int{left + 1, right + 1}
        } else if sum < target {
            left++
        } else {
            right--
        }
    }

    return []int{0, 0}
}

File: 0167-two-sum-ii---input-array-is-sorted/solution.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers) - 1

        while l < r:
            left = numbers[l]
            right = numbers[r]
            if left + right > target:
                r -= 1
            elif left + right == target:
                return [l + 1, r + 1]
            else:
                l += 1
        
        return [-1, -1]
